sklearn/calibration.py

```diff
@@ -656,10 +656,10 @@ def predict_proba(self, X):
         if n_classes == 2:
             proba[:, 0] = 1. - proba[:, 1]
         else:
-            proba /= np.sum(proba, axis=1)[:, np.newaxis]
-
-        # XXX : for some reason all probas can be 0
-        proba[np.isnan(proba)] = 1. / n_classes
+            denominator = np.sum(proba, axis=1)[:, np.newaxis]
+            # XXX : for some reason all probas can be 0
+            proba = np.divide(proba, denominator, out=np.full_like(
+                proba, 1 / n_classes), where=denominator != 0)
```

Comment:
I would find it a bit more readable to avoid splitting the `np.full_like` expression on two lines:

```suggestion
            uniform_proba = np.full_like(proba, 1 / n_classes)
            proba = np.divide(proba, denominator, out=uniform_proba,
                              where=denominator != 0)
```